vulture_os/applications/logfwd/config/om_elasticsearch_pre.tpl

```diff
@@ -0,0 +1,88 @@
+{%- if retry_on_els_failures %}
+ruleset(name="{{output_name}}_retry_failure") {
+    action(type="omfile" template="RSYSLOG_DebugFormat" file="/var/log/internal/{{output_name}}_error.log")
+}
+{%- endif -%} {# if retry_on_els_failures #}
+
+ruleset(name="{{output_name}}_retry") {
+    if strlen($.omes!status) > 0 then {
+        # retry case
+        if ($.omes!status == 200) or ($.omes!status == 201) or (($.omes!status == 409) and ($.omes!writeoperation == "create")) then {
+            stop # successful
+        }
+        if ($.omes!writeoperation == "unknown") or (strlen($.omes!error!type) == 0) or (strlen($.omes!error!reason) == 0) then {
+            call {{output_name}}_retry_failure
+            stop
+        }
+        if ($.omes!status == 400) or ($.omes!status < 200) then {
+            call {{output_name}}_retry_failure
+            stop
+        }
```

Comment:
cannot call the {{output_name}}_retry_failure ruleset if it's conditionaly created above!

Reply:
I have correctly move the condition to hide both rulesets if failure handling is unchecked.
I don't do it in the models in case we add other features in the future.
Free to debate about it.